helper/schema/schema.go

```diff
@@ -1049,6 +1059,12 @@ func (m schemaMap) validateList(
 			"%s: should be a list", k)}
 	}
 
+	// Validate length
+	if schema.MaxItems > 0 && rawV.Len() > schema.MaxItems {
+		return nil, []error{fmt.Errorf(
+			"%s: exceeded MaxItems of %d", k, schema.MaxItems)}
```

Comment:
Since this error message will end up being user facing, let's make the language verbose so the user understands exactly what they need to fix.

Something like (w/ an in-context example):

```
aws_instance.foo: root_block_device: attribute supports 1 item maximum, config has 2 declared
```

Maybe we can think of some clearer language, but just something to help users correct their config when they see the error is what I'm getting at.